src/main/java/Duke.java

```diff
@@ -1,10 +1,241 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
+    /** count tracker for number of tasks in array */
+    private static int taskCount = 0;
+
+    /** count tracker for number of uncompleted tasks in array */
+    private static int numOfUncompletedTasks = 0;
+
+    /** Array of tasks */
+    private static final Task[] tasks = new Task[100];
+
+    public static void greeting() {
+        System.out.println("_____________________________________________________");
+        System.out.println("Hello! I'm Duke the Dancing Dragon.");
+        System.out.println("Anything I can help you with, young Padawan?");
+        System.out.println("_____________________________________________________");
+    }
+
+    public static void goodbyeMessage() {
+        System.out.println("It's over Anakin... I can finally eat my lun-");
+        System.out.println("_____________________________________________________");
+    }
+
+    public static void printDone(Task arg) {
+        System.out.println("You have marked item " + arg.description + " as done:");
+        System.out.println(arg.getStatusIcon() + " " + arg.description);
+    }
+
+    /**
+     * Returns the list number which is assigned to the task.
+     *
+     * @param arg user input that contains word [done].
+     * @return task number.
+     */
+    public static int getTaskNum(String arg) {
+        String[] words = arg.trim().split("[\\s]+");
+        return Integer.parseInt(words[1]);
+    }
+
+    /**
+     * Returns the description of the task only, without the date or the keyword.
+     *
+     * @param query user raw data input.
+     * @return description of task.
+     */
+    public static String getQueryDescription(String query) {
+        String[] words = query.trim().split("[\\s]+");
+        String[] allButFirstWord = Arrays.copyOfRange(words, 1, words.length);
+        StringBuilder sentenceAfterDeletion = new StringBuilder();
+        for (String word : allButFirstWord) {
+            if (word.contains("/")) {
+                break;
+            } else {
+                sentenceAfterDeletion.append(word).append(" ");
+            }
+        }
+        return sentenceAfterDeletion.toString();
+    }
+
+    /**
+     * Returns date value for tasks which need a date input field.
+     * If user does not have proper date formatting, (i.e. '/by' or '/at') this function returns an empty string.
+     *
+     * @param query user raw data input.
+     * @return date value
+     */
+    public static String getDate(String query) {
+        int slashPosition = query.indexOf("/");
+        if (slashPosition == -1) {
+            return "";
+        } else {
+            int datePosition = slashPosition + 3;
```

Comment:
Same problem here, maybe the 3 can be assigned to a global constant to prevent it from being a magic literal?